tck/src/main/java/org/eclipse/microprofile/lra/tck/service/spi/LRARecoveryService.java

```diff
@@ -0,0 +1,80 @@
+ */
+public interface LRARecoveryService {
+
+    /**
+     * This method is utilized by the TCK to wait for the delivery of all participant callbacks
+     * (Complete, Compensate, AfterLRA, ...) for a given LRA. When the LRA is asked to perform some enclosing
+     * eventually consistent action (close/cancel) which results into subsequent calls to every Complete/Compensate
+     * method of individual enlisted participants and this method is invoked it shouldn't return before
+     * all of the responses from the participant callbacks are successfully received. When this method returns
+     * the participant methods invocation results are verified so the implementation must make sure that this method
+     * doesn't return before the validation can happen.
+     * <p/>
+     * 
+     * Example:
+     * <ol>
+     *   <li>call to close LRA with id lraId (Complete calls are starting to be performed)</li>
+     *   <li>call to {@link LRARecoveryService#waitForCallbacks(URI)} with lraId -- wait for all Complete calls to be
+     * delivered</li>
+     *   <li>call to {@link LRARecoveryService#waitForCallbacks(URI)} returns</li>
+     *   <li>the TCK test verifies that all Complete methods have been called</li>
+     * </ol>
+     *
+     * @param lraId the LRA id of the LRA which callbacks should be delivered when this method exits
+     */
+    void waitForCallbacks(URI lraId);
+
+    /**
+     * This method is utilized by the TCK to wait for the recovery of a particular LRA
+     * (replay of Status, Complete, Compensate) once. This means that this method should be called when the passed
+     * LRA is in one of its intermediate states (Completing, Compensating). The invocation should retry the calls
+     * to all enlisted participant methods according to the specification and wait for their completion (meaning the
+     * delivery of the participants responses back to the implementation). When this method returns the participant
+     * methods invocations results are verified so the implementation must make sure that the method doesn't return
+     * before the validation can happen.
+     * <p/>
+     * 
+     * Example:
+     * <ol>
+     *   <li>call to close LRA with id lraId (Complete calls are starting to be performed)</li>
+     *   <li>one or more of the enlisted participants returns HTTP status code 202 from its
+     *   completion notification which implies the participant is Completing</li>
+     *   <li>call to {@link LRARecoveryService#waitForRecovery(URI)} with the lraId -- trigger immediate replay
+     * of the Complete (or Status) call and wait for its delivery</li>
+     *   <li>call to {@link LRARecoveryService#waitForRecovery(URI)} returns</li>
+     *   <li>the TCK test verifies that Complete (or Status) have been called (retried)</li>
+     * </ol>
+     * 
+     * @param lraId the LRA id of the LRA which should be recovered when this method exits
+     */
+    void waitForRecovery(URI lraId);
```

Comment:
so `waitForRecovery` is not a good name is it actively needs to perform things.

Reply:
can (and should) but doesn't have to. Also long delay would work here.